FirebaseSessions/Sources/Settings/RemoteSettings.swift

```diff
@@ -102,10 +114,12 @@ extension RemoteSettingsConfigurations {
   }
 
   func isSettingsStale() -> Bool {
-    return isCacheExpired(time: Date())
+    cache.withLock { cache in
+      isCacheExpired(cache, time: Date())
+    }
   }
 
-  private func isCacheExpired(time: Date) -> Bool {
+  private func isCacheExpired(_ cache: SettingsCacheClient, time: Date) -> Bool {
```

Comment:
I wrote this this way to work around reentrancy issues with the `cache` variable once I put it behind a lock, since it's called in the critical section below. Maybe it'd be good to document.

Reply:
Thanks, I see the issue. This method looks to me like it should be an instance method anyhow. It references more members of the cache class than the current one.

Reply:
Approach in 1c3797a5a7065242e93df758dee422fb11a0c32a

The API of SettingsCacheClient should probably change to be internally thread-safe, but this should be okay for now.